diff --git a/programs/staking/src/lib.rs b/programs/staking/src/lib.rs
--- a/programs/staking/src/lib.rs
+++ b/programs/staking/src/lib.rs
@@ -22,6 +22,10 @@ pub mod staking {
     pub fn close_stake(ctx: Context<CloseStake>) -> Result<()> {
         let dest = ctx.accounts.destination.to_account_info();
         let src = ctx.accounts.stake_account.to_account_info();
+        **dest.lamports.borrow_mut() += **src.lamports.borrow();
+        **src.lamports.borrow_mut() = 0;
         Ok(())
     }
 }
